### DISCUSSION TITLE
Encrypt file with public key only which doesn't have expiry date

### DISCUSSION THREAD

**Discussion by juris-kreilis** [pgpainless/pgpainless#402] [Q&A] [answered]
Hello,

I am trying to encrypt a simple file with an existing public key which has no expiration date.  Tried  with [pgpainless-sop](https://pgpainless.readthedocs.io/en/latest/quickstart.html#encrypt-a-message) as well
```
 PGPPublicKeyRing publicKeyRing = PGPainless.readKeyRing().publicKeyRing(publicKeyContent);

            ByteArrayOutputStream ciphertext = new ByteArrayOutputStream();
            EncryptionStream encryptor = PGPainless.encryptAndOrSign()
                    .onOutputStream(ciphertext)
                .withOptions(ProducerOptions.encrypt(EncryptionOptions.encryptCommunications().addRecipient(publicKeyRing)).setAsciiArmor(true));
```

I am getting `Key XXXXXXXXXXXXXXXX does not have a valid/acceptable signature to derive an expiration date from`. exception. 

Some GUI tools like PGPTool or others allows to do this. Is this a defect or a specific limitation?

**Comment by vanitasvitae** [comment, marked as answer]
Pretty sure you are running into https://github.com/pgpainless/pgpainless/issues/398

**Reply by juris-kreilis** [reply]
Thanks, it helped!
